components/automate-ui/src/app/modules/infra-proxy/reset-client-key/reset-client-key.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { takeUntil } from 'rxjs/operators';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { combineLatest, Subject } from 'rxjs';
+import { ResetKeyClient } from 'app/entities/clients/client.action';
+import { getStatus,
+  resetKeyClient,
+  saveError } from 'app/entities/clients/client-details.selectors';
+import { EntityStatus } from 'app/entities/entities';
+import { isNil } from 'lodash/fp';
+import { saveAs } from 'file-saver';
+
+@Component({
+  selector: 'app-reset-client-key',
+  templateUrl: './reset-client-key.component.html',
+  styleUrls: ['./reset-client-key.component.scss']
+})
+
+export class ResetClientKeyComponent implements OnInit, OnDestroy {
+  @Input() openEvent: EventEmitter<boolean>;
+  @Input() serverId: string;
+  @Input() orgId: string;
+  @Input() name: string;
+
+  public authFailure = false;
+  public isReset = false;
+  public reseting = false;
+  public conflictError = false;
+  public error: string;
+  public publicKey: string;
+  public privateKey: string;
+  public org: string;
+  public server: string;
+  public visible = false;
+  public conflictErrorEvent = new EventEmitter<boolean>();
+  public close = new EventEmitter();
+  private isDestroyed = new Subject<boolean>();
+
+  constructor(
+    private store: Store<NgrxStateAtom>
+  ) { }
+
+  ngOnInit() {
+    this.openEvent.pipe(takeUntil(this.isDestroyed))
+      .subscribe(() => {
+        this.conflictError = false;
+        this.visible = true;
+        this.reseting = false;
+        this.isReset = false;
+        this.server = this.serverId;
+        this.org = this.orgId;
+        this.name = this.name;
+        this.error = '';
+        this.privateKey = '';
+      });
+
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(resetKeyClient),
+      this.store.select(saveError)
+    ]).pipe(
+      takeUntil(this.isDestroyed))
+      .subscribe(([getStatusSt, resetKeyState, errorSt]) => {
+      if (getStatusSt === EntityStatus.loadingSuccess &&
+        !isNil(resetKeyState)) {
+            this.reseting = false;
+            this.isReset = true;
+            this.conflictError = false;
+            this.privateKey = resetKeyState?.client_key.private_key;
+          } else if (getStatusSt === EntityStatus.loadingFailure) {
+            this.error = errorSt;
+            this.closeCreateModal();
+            this.authFailure = true;
+          }
+        });
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  public handleInput(event: KeyboardEvent): void {
```

Comment:
Let's be consistent -- omit the qualifier for all public methods. (Private methods should always have `private`.)
```suggestion
  handleInput(event: KeyboardEvent): void {
```